orderbook/src/database/orders.rs

```diff
@@ -56,6 +57,30 @@ impl DbOrderKind {
     }
 }
 
+/// Location for which the sellAmount should be drawn upon order fulfilment
+#[derive(sqlx::Type)]
+#[sqlx(type_name = "DbBalanceFrom")]
+#[sqlx(rename_all = "snake_case")]
+pub enum DbBalanceFrom {
+    /// Direct ERC20 allowances to the Vault relayer contract
+    Erc20,
+    /// ERC20 allowances to the Vault with GPv2 relayer approval
+    Internal,
+    /// Internal balances to the Vault with GPv2 relayer approval
+    External,
```

Comment:
If you want to use `Vault*` in the Rust type, you *may* be able to use something like:
```suggestion
    /// ERC20 allowances to the Vault with GPv2 relayer approval
    #[sqlx(rename = "internal")]
    VaultInternal,
    /// Internal balances to the Vault with GPv2 relayer approval
    #[sqlx(rename = "external")]
    VaultExternal,
```

I agree that the names aren't great - maybe we should revive the discussion https://github.com/gnosis/gp-v2-contracts/issues/543

I'm also 100% fine as is, but just trying to make a suggestion that would improve the naming in the Rust codebase, while still keeping the database values consistent with what was signed by the user.

Reply:
I think we can leave it as is for now and I have started to become used to it. The naming now suggests that the Balancer integration is so tight it actually gets the default naming convention by excluding the term Vault. Anyway, this can be changed later.